pkg/pub_package_reader/lib/src/package_dir.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'tar_utils.dart';
+
+/// Helper methods to access package content.
+abstract class PackageDir {
+  /// List all the file names in the directory.
+  Future<List<String>> listFileNames();
+
+  /// Whether the file on the given relative [path] exists.
+  Future<bool> exists(String path);
```

Comment:
What happens for folders? symlinks? etc?

Reply:
I think that's a fair concern, but I'm not sure how should I add to this part. Should it return true only if the file (and parent directories) are not symlinked?